lib/htmltoword/xslt/image_functions.xslt

```diff
@@ -0,0 +1,148 @@
+                xmlns:fn="http://www.w3.org/2005/xpath-functions"
+                version="1.0"
+                exclude-result-prefixes="java msxsl ext w o v WX aml w10"
+                extension-element-prefixes="func">
+
+  <!-- template as function used to return the name of an image. -->
+  <xsl:template name="image-extension">
+    <xsl:param name="data-filename" select="." />
+    <xsl:param name="source" select="." />
+    <xsl:variable name="filename">
+      <xsl:call-template name="image-name">
+        <xsl:with-param name="source" select="$source"/>
+        <xsl:with-param name="data-filename" select="$data-filename"/>
+      </xsl:call-template>
+    </xsl:variable>
+    <xsl:value-of select="substring-after($filename,'.')"/>
+  </xsl:template>
+
+  <!-- template as function used to return the name of an image. -->
+  <xsl:template name="image-name">
+    <xsl:param name="data-filename" select="." />
+    <xsl:param name="source" select="." />
+    <xsl:choose>
+      <xsl:when test="string-length($data-filename) > 0">
+        <xsl:value-of select="$data-filename"/>
+      </xsl:when>
+      <xsl:otherwise>
+        <xsl:call-template name="extract-filename-from-path">
+          <xsl:with-param name="path" select="$source"/>
+        </xsl:call-template>
+      </xsl:otherwise>
+    </xsl:choose>
+  </xsl:template>
+
+  <!-- template as function used to extract the filename from the image source. Can't use tokenize or other functions which return a fragment tree as xpath can't process them. -->
+  <xsl:template name="extract-filename-from-path">
+      <xsl:param name="path" select="." />
+      <xsl:choose>
+        <xsl:when test="not(contains($path, '/'))">
+            <xsl:value-of select="$path" />
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:call-template name="extract-filename-from-path">
+            <xsl:with-param name="path" select="substring-after($path, '/')" />
+          </xsl:call-template>
+        </xsl:otherwise>
+      </xsl:choose>
+  </xsl:template>
+
+  <!--
+    Returns the unqualified dimension from a length specification copied from:
+  http://docbook.sourceforge.net/release/xsl/1.76.1/doc/lib/length-magnitude.html
+   -->
+
+  <xsl:template name="length-magnitude">
+    <xsl:param name="length" select="'0pt'"></xsl:param>
+
+    <xsl:choose>
+      <xsl:when test="string-length($length) = 0"></xsl:when>
+      <xsl:when test="substring($length,1,1) = '0'                     or substring($length,1,1) = '1'                     or substring($length,1,1) = '2'                     or substring($length,1,1) = '3'                     or substring($length,1,1) = '4'                     or substring($length,1,1) = '5'                     or substring($length,1,1) = '6'                     or substring($length,1,1) = '7'                     or substring($length,1,1) = '8'                     or substring($length,1,1) = '9'                     or substring($length,1,1) = '.'">
```

Comment:
No need, just helped to break up the numerous conditions though I appreciate they aren't aligned here. 

Shall I line them up? I'm loathed to take out all of the spaces as then it will be impossible to read...